### ISSUE TITLE
multibit classic takes days to syncrhonize?

### ISSUE THREAD

**Issue by ZatoshiBrother** [bitcoin-solutions/multibit#734]
'm with the latest version of MultiBit Classic for linux and it spent the last day trying to syncrhonize and is still an year behind. I thought this is thin client? Is this normal or there is something I should fix. I can NOT migrate to MultiBit HD at the moment. Need to fix this one.

**Comment by jim618** [comment]
If you have several wallets and/ or lot of addresses (hundreds or more) in Classic then yes you will see this behaviour unfortunately.

Classic is not being actively developed so this is unlikely to get fixed. We are keeping the code base stable now.

If you have several wallets you can have several instances of MultiBit, each with a single wallet open.
That helps. If you look in the help for how to run from a USB drive (but just put MultiBit and your wallet somewhere on disk) it explains how to do it.

It that doesn't work you could move your wallet(s) to blockchain.info where it will be quicker.
There is a how-to in the help on how to do this.

**Comment by ZatoshiBrother** [comment]
Yes exactly. I have several wallets with hundreds of addresses in each. 
Ok I'll just wait then.

By the way one of the reasons I hold the migration to HD is because it doesn't seem like it's interface is made to work easily with bunch of different wallets at once in one intance? 
That's why I like Classic so much.

**Comment by traeh22** [comment]
You can't export a private key while Multibit Classic is syncing. I have been syncing for over 24 hours now. How am I supposed to get my private key to export it to a new program while its stuck in sync?

**Comment by jim618** [comment]
There isn't a way to perform "wallet level" operations like exporting/ importing etc whilst the wallet is syncing.
This is mainly because of the threading in the app - whilst syncing is happening there are a lot of threads interacting with the wallet.

The main determinants with sync speed are:
+ the total number of addresses you have (in all the wallets you have open)
+ the speed of your network. MB/s level internet is what you want.

@ZatoshiBrother Yes having multiple wallets in Classic is useful but we found it complicated the UI and makes 'there are now lots of addresses to sync' appear sooner. In HD each wallet is independent and syncs separately.
